lib/tokenize.cpp

```diff
@@ -3424,14 +3424,85 @@ void Tokenizer::simplifyTemplates()
 //---------------------------------------------------------------------------
 
 
-static bool setVarIdParseDeclaration(const Token **tok, const std::map<std::string,int> &variableId, bool executableScope, bool cpp, bool c)
+/** Class used in Tokenizer::setVarIdPass1 */
+class VariableMap {
+private:
+    std::map<std::string, nonneg int> mVariableId;
+    std::map<std::string, nonneg int> mVariableId_global;
```

Comment:
hmm I don't feel 100% sure about this code and I guess you also feel that way. But `mVariableId_global` will contain the same strings and varIds as first/bottom item in `mScopeInfo` right? So I am thinking that technically it might be redundant..

Reply:
Sorry for the late reply, I thought I did already reply to this.

Please note that this is all temporary data which will be freed after setVarId is finished. mVariableId_global is necessary to define which variables are accessible at global scope (with operator ::) even though the code is in some local scope. Yes, in some way it is redundant information, but I think it is easier to have this temporary copy to have a similar mechanism to retrieve the varids for local and global scope instead of searching inside the std::vector of std::pair (which is not designed for searching). Alternatively, more of the code has to be rewritten (replace the vector of pairs).